src/Connection.php

```diff
@@ -1283,16 +1283,17 @@ public function transactional(Closure $func)
 
         try {
             $res = $func($this);
-            $this->commit();
 
             $successful = true;
-
-            return $res;
         } finally {
             if (! $successful) {
                 $this->rollBack();
             }
         }
+
+        $this->commit();
```

Comment:
What if `commit` throws for some other reason, before anything is transmitted to the database?

Reply:
Can you describe some scenario you have in mind so I can cover it with test?

Reply:
Here, there are throws statements:

https://github.com/doctrine/dbal/blob/220aec46be990c0900826e29d9e3faa53bfbf70a/src/Connection.php#L1013-L1026

I don't know if there is a realistic scenario in which they can throw when `commit` is called from the method you are modifying, before the transaction is actually committed, but that's what I was thinking about.

Reply:
I see. Those should not happen but I have adjusted the flow and covered with unit test should it happen.

cf80105b07b8f46e19c56e71fdff826ecbc0c4fa